rest_api/controller/feedback.py

```diff
@@ -102,86 +63,107 @@ def eval_doc_qa_feedback(filters: FilterRequest = None):
     SQuAD format JSON export for question/answer pairs that were marked as "relevant".
 
     The context_size param can be used to limit response size for large documents.
     """
-    labels = document_store.get_all_labels(
-        index=DB_INDEX_FEEDBACK, filters={"is_correct_answer": [True], "origin": ["user-feedback"]}
-    )
+    if only_positive_labels:
+        labels = document_store.get_all_labels(filters={"is_correct_answer": [True], "origin": ["user-feedback"]})
+    else:
+        labels = document_store.get_all_labels(filters={"origin": ["user-feedback"]})
+        # Filter out the labels where the passage is correct but answer is wrong (in SQuAD this matches
+        # neither a "positive example" nor a negative "is_impossible" one)
+        labels = [l for l in labels if not (l.is_correct_document is True and l.is_correct_answer is False)]
 
     export_data = []
-    for label in labels:
-        document = document_store.get_document_by_id(label.document_id)
-        text = document.text
 
-        # the final length of context(including the answer string) is 'context_size'.
-        # we try to add equal characters for context before and after the answer string.
-        # if either beginning or end of text is reached, we correspondingly
-        # append more context characters at the other end of answer string.
-        context_to_add = int((context_size - len(label.answer)) / 2)
-
-        start_pos = max(label.offset_start_in_doc - context_to_add, 0)
-        additional_context_at_end = max(context_to_add - label.offset_start_in_doc, 0)
-
-        end_pos = min(label.offset_start_in_doc + len(label.answer) + context_to_add, len(text) - 1)
-        additional_context_at_start = max(label.offset_start_in_doc + len(label.answer) + context_to_add - len(text), 0)
-
-        start_pos = max(0, start_pos - additional_context_at_start)
-        end_pos = min(len(text) - 1, end_pos + additional_context_at_end)
-
-        context_to_export = text[start_pos:end_pos]
-
-        export_data.append({"paragraphs": [{"qas": label, "context": context_to_export}]})
-
-    export = {"data": export_data}
-
-    return export
-
-
-@router.get("/export-faq-qa-feedback")
-def export_faq_feedback():
-    """
-    Export feedback for faq-qa in JSON format.
-    """
-
-    labels = document_store.get_all_labels(index=DB_INDEX_FEEDBACK, filters={"origin": ["user-feedback-faq"]})
-
-    export_data = []
     for label in labels:
         document = document_store.get_document_by_id(label.document_id)
-        feedback = {
-            "question": document.question,
-            "query": label.question,
-            "is_correct_answer": label.is_correct_answer,
-            "is_correct_document": label.is_correct_answer,
-        }
-        export_data.append(feedback)
+        if document is None:
+            raise HTTPException(
+                status_code=500, detail="Could not find document with id {label.document_id} for label id {label.id}"
```

Comment:
just nip: `HTTPStatus.NOT_FOUND` status would be more suitable here.

Reply:
HTTP 4x error codes are client-side, i.e., a client requesting a resource with an ID that does not exist. This means that the client can ensure the correctness of the resource identifier and try again.

In this case, it's a server-side error where data is inconsistent, i.e., a label has an incorrect document ID.

Reply:
I misunderstood that doc id is user supplied. We can replace this with `HTTPStatus .INTERNAL_SERVER_ERROR` for more readability.